### PR TITLE
Wrap up CPDAcknowledgements for release

### PR DESCRIPTION
**Pull request by orta**
- [x] Uses Nullable and lightweight generics
- [x] Tightens the UI a bit more
- [x] Adds docs to all the public info
- [x] Uses CP 1.0 in the example
- [x] Update README, provide some example images

### CODE REVIEW

**Comment by orta** [comment]
Closes https://github.com/orta/life/issues/60

File: CPDAcknowledgements.podspec [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
   s.description  = <<-DESC
-                    Show your CocoaPods library and contributors in-app with smart defaults, and view controllers.
+                    Show your CocoaPods library and contributors in-app with smart defaults, and customisable view controllers.
                    DESC
   s.homepage     = "https://github.com/CocoaPods/CPDAcknowledgements"
-  s.screenshots  = "www.example.com/screenshots_1", "www.example.com/screenshots_2"
   s.license      = 'MIT'
   s.author       = { "Orta Therox" => "orta.therox@gmail.com", "Fabio Pelosin" => "fabiopelosin@gmail.com" }
   s.source       = { :git => "https://github.com/CocoaPods/CPDAcknowledgements.git", :tag => s.version.to_s }
+  s.homepage      = "https://cocoapods.org"
```
Comment (segiddins) [inline CPDAcknowledgements.podspec]: Should probably point to the github repo for now

File: CPDAcknowledgements.podspec [hunk 2]
```diff
@@ -1,20 +1,19 @@
 Pod::Spec.new do |s|
   s.name         = "CPDAcknowledgements"
-  s.version      = "0.0.1"
+  s.version      = "0.5.0"
```
Comment (segiddins) [inline CPDAcknowledgements.podspec]: Why not 1.0?

File: CPDAcknowledgements.podspec [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
   s.author       = { "Orta Therox" => "orta.therox@gmail.com", "Fabio Pelosin" => "fabiopelosin@gmail.com" }
   s.source       = { :git => "https://github.com/CocoaPods/CPDAcknowledgements.git", :tag => s.version.to_s }
+  s.homepage      = "https://cocoapods.org"
   s.social_media_url = "https://twitter.com/CocoaPods"
-  s.platform     = :ios, '6.0'
-  s.ios.deployment_target = '6.0'
-  s.requires_arc = true
-  s.source_files = 'Classes'
-  s.ios.source_files = 'Classes/ios'
+  s.platform     = :ios, '8.0'
```
Comment (segiddins) [inline CPDAcknowledgements.podspec]: This line should be deleted

File: CPDAcknowledgements.podspec [hunk 4]
```diff
@@ -1,20 +1,19 @@
...
+  s.ios.deployment_target = '8.0'
+  s.source_files = 'CPDAcknowledgements/**/**'
```
Comment (segiddins) [inline CPDAcknowledgements.podspec]: This won't match anything, you want `**/*`

File: CPDAcknowledgements/CPDAcknowledgementsViewController.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+/// installed and running on a `pod install`, and will pull all the library metadata from that.
+/// You can pass in a collection of `CPDContribution`s to provide individual contributions outside of libraries.
+/// The style can be used to customize the look and feel of a library's information page.
+
+- (instancetype)initWithStyle:(CPDStyle * _Nullable )style acknowledgements:(NSArray <CPDLibrary *>* _Nullable )acknowledgements contributions:(NSArray <CPDContribution *>* _Nullable )contributions;
+
+@end
+
+
+NS_ASSUME_NONNULL_END
```
Comment (segiddins) [inline CPDAcknowledgements/CPDAcknowledgementsViewController.h]: trailing newline

File: CPDAcknowledgements/CPDAcknowledgementsViewController.m [hunk 6]
```diff
@@ -70,33 +62,43 @@ - (void)configureAcknowledgements:(NSArray *)acknowledgements contributions:(NSA
...
-
     self.tableView.dataSource = self.dataSource;
 }
 
+- (void)viewWillAppear:(BOOL)animated
+{
+    [super viewWillAppear:animated];
+
+    if(!self.navigationController){
+        @throw @"The AcknowledgementVC needs to be inside a navigation controller.";
```
Comment (segiddins) [inline CPDAcknowledgements/CPDAcknowledgementsViewController.m]: imho should just be `NSAssert(self.navigationController)`

File: CPDAcknowledgements/CPDContribution.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+/// Responsibility in their part of shipping your app
+@property (nonatomic, copy, readonly) NSString *role;
+
+/// Optional avatar url which will be grabbed asynchronously
+@property (nonatomic, copy) NSString * _Nullable avatarAddress;
+
+@end
+
+NS_ASSUME_NONNULL_END
```
Comment (segiddins) [inline CPDAcknowledgements/CPDContribution.h]: trailing newline

File: CPDAcknowledgements/CPDLibrary.h [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+/// Perform actions from given titles
+- (void)performActionAtIndex:(NSInteger)index;
+
+/// Is it worth showing the popover based on the current available metadata?
+- (BOOL)hasActions;
+
+@end
+
+NS_ASSUME_NONNULL_END
```
Comment (segiddins) [inline CPDAcknowledgements/CPDLibrary.h]: trailing newline

File: CPDAcknowledgements/CPDLibrary.m [hunk 9]
```diff
@@ -61,11 +51,14 @@ - (NSString *)twitterHandle
     return nil;
 }
 
+- (BOOL)hasActions
+{
+    return (self.websiteAddress.length > 0 || [self.socialMediaAddress hasPrefix:@"https://twitter"]);
```
Comment (segiddins) [inline CPDAcknowledgements/CPDLibrary.m]: the parens are unecessary

File: CPDAcknowledgements.podspec (hunk 4 above)
Comment (orta) [inline CPDAcknowledgements.podspec]: It does, that's OK :+1:

**Comment by orta** [comment]
Thanks @segiddins - all addressed, I'm gonna give it a quick one over, then shippit

**Pull request merged**